source/fem-dem/gls_vans.cc

```diff
@@ -663,56 +817,52 @@ GLSVANSSolver<dim>::quadrature_centered_sphere_method(bool load_balance_step)
 
           // Define the volume of the reference sphere to be used as the
           // averaging volume for the QCM
-          double reference_sphere_volume;
-
-          if (this->cfd_dem_simulation_parameters.void_fraction
-                ->qcm_sphere_diameter < 1e-16)
+          if (calculate_reference_sphere_radius)
             {
-              // Volume of sphere equals volume of cell
               if (this->cfd_dem_simulation_parameters.void_fraction
                     ->qcm_sphere_equal_cell_volume == true)
-                reference_sphere_volume = cell->measure();
+                {
+                  // Get the radius by the volume of sphere which is
+                  // equal to the volume of cell
+                  R_sphere =
+                    0.5 * pow(2 * dim * cell->measure() / M_PI, 1. / dim);
+                }
               else
-                // Volume of sphere based on R_s = h_omega
-                reference_sphere_volume =
-                  M_PI * pow((2 * pow(cell->measure(), 1. / dim)), dim) /
-                  (2 * dim);
-            }
-          else
-            {
-              // Volume of sphere based on R_s = user defined
-              reference_sphere_volume =
-                M_PI * pow((qcm_sphere_diameter), dim) / (2 * dim);
+                {
+                  // The radius is obtained from the volume of sphere based
+                  // on R_s = h_omega
+                  double reference_sphere_volume =
+                    M_PI * pow((2 * pow(cell->measure(), 1. / dim)), dim) /
+                    (2 * dim);
+                  R_sphere = 0.5 * pow(2 * dim * reference_sphere_volume / M_PI,
+                                       1. / dim);
```

Comment:
Shouldnt it be that the radius is the diameter of that sphere instead of its radius?
It seems a bit pessimistic to define it using the radius? Maybe we can revisit this in a next PR

Reply:
You mean that the idea of this radius definition may be wrong or the modification is wrong?